controllers/volumereplicationgroup_controller.go

```diff
@@ -1142,29 +1204,31 @@ func (v *VRGInstance) undoPVRetentionForPVC(pvc corev1.PersistentVolumeClaim, lo
 }
 
 // Upload PV to the list of S3 stores in the VRG spec
-func (v *VRGInstance) uploadPVToS3Profiles(pvc corev1.PersistentVolumeClaim) (err error) {
+func (v *VRGInstance) uploadPVToS3Stores(pvc corev1.PersistentVolumeClaim) (err error) {
 	for _, s3ProfileName := range v.instance.Spec.S3ProfileList {
-		if err := v.uploadPV(s3ProfileName, pvc); err != nil {
+		if err := v.reconciler.PVUploader.UploadPV(v, s3ProfileName, pvc); err != nil {
 			return fmt.Errorf("error uploading PV %s using profile %s, err %w", pvc.Name, s3ProfileName, err)
 		}
 	}
 
 	return
 }
 
-// uploadPV checks if the VRG spec has been configured with an s3 endpoint,
+type ObjectStorePVUploader struct{}
```

Comment:
Why do we need an empty struct to act as a class?

Reply:
In order to implement the interface method UploadPV. At runtime, we can pass a fake uploader or real uploader depending on where it is instantiated from.  For the unit test, the fake is instantiated in the suite_test.go and for production, the real one is instantiated in main.go